app/api/items/add_item.go

```diff
@@ -3,145 +3,292 @@ package items
+		TeamMode        *string    `json:"team_mode" sql:"column:sTeamMode" validate:"oneof=All Half One None"`
+		TeamInGroupID   *int64     `json:"team_in_group_id" sql:"column:idTeamInGroup" validate:"team_in_group_id"`
+		TeamMaxMembers  int32      `json:"team_max_members" sql:"column:iTeamMaxMembers"`
+		TitleBarVisible bool       `json:"title_bar_visible" sql:"column:bTitleBarVisible"`
+		HasAttempts     bool       `json:"has_attempts" sql:"column:bHasAttempts"`
+		AccessOpenDate  *time.Time `json:"access_open_date" sql:"column:sAccessOpenDate"`
+		Duration        *string    `json:"duration" sql:"column:sDuration" validate:"duration"`
+		EndContestDate  *time.Time `json:"end_contest_date" sql:"column:sEndContestDate"`
+		ContestPhase    *string    `json:"contest_phase" sql:"column:sContestPhase" validate:"oneof=Running Analysis Closed"`
+		ShowUserInfos   bool       `json:"show_user_infos" sql:"column:bShowUserInfos"`
+		Level           *int32     `json:"level" sql:"column:iLevel"`
+		UsesAPI         bool       `json:"uses_api" sql:"column:bUsesAPI"`
+		GroupCodeEnter  *bool      `json:"group_code_enter" sql:"column:groupCodeEnter"`
+	} `json:"item" sql:"column:items"`
+	LanguageID int64 `json:"language_id" validate:"required,language_id"`
+	String     struct {
+		Title       string  `json:"title" validate:"required" sql:"column:sTitle"`
+		ImageURL    *string `json:"image_url" sql:"column:sImageUrl"`
+		Subtitle    *string `json:"subtitle" sql:"column:sSubtitle"`
+		Description *string `json:"description" sql:"column:sDescription"`
+	} `json:"string"`
+
+	ParentItemID int64 `json:"parent_item_id,string" validate:"required,parent_item_id"`
+	Order        int32 `json:"order"`
+
+	Children []itemChild `json:"children" validate:"children"`
 }
 
-func (in *NewItemRequest) itemData() *database.Item {
-	return &database.Item{
-		ID:                in.ID.Int64,
-		Type:              in.Type.String,
-		DefaultLanguageID: in.Strings[0].LanguageID.Int64,
-		TeamsEditable:     *types.NewBool(false), // has no db default at the moment, so must be set
-		NoScore:           *types.NewBool(false), // has no db default at the moment, so must be set
+func (in *NewItemRequest) itemData(itemID int64, itemMap map[string]interface{}) map[string]interface{} {
+	itemMap["ID"] = itemID
+	itemMap["idDefaultLanguage"] = in.LanguageID
+	if _, ok := itemMap["sContestPhase"]; !ok {
+		itemMap["sContestPhase"] = "Running"
 	}
+	return itemMap
 }
 
-func (in *NewItemRequest) groupItemData(id, userID, groupID int64) *database.GroupItem {
-	return &database.GroupItem{
-		ID:             *types.NewInt64(id),
-		ItemID:         in.ID.Int64,
-		GroupID:        *types.NewInt64(groupID),
-		CreatorUserID:  *types.NewInt64(userID),
-		FullAccessDate: *types.NewDatetime(time.Now()),
-		OwnerAccess:    *types.NewBool(true),
-		ManagerAccess:  *types.NewBool(true),
+func (in *NewItemRequest) groupItemData(groupItemID, userID, groupID, itemID int64) map[string]interface{} {
```

Comment:
When you read these functions alone, it is not easy to understand what it stands for. It is clearer when you see their call. You should add a simple comment such as "Creating the map containing the db data to be inserted into the GroupItem table" etc

Reply:
fixed